### PR TITLE
Refactor JobQueueNode._monitor

### PR DESCRIPTION
**Pull request by eivindjahren**
## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/job_queue/job_queue_node.py [hunk 1]
```diff
@@ -183,73 +215,85 @@ def runtime(self) -> float:
...
-                    )
-                elif self._tried_killing % 100 == 0:
-                    logger.warning(
-                        f"Tried killing with MAX_RUNTIME {self._tried_killing} "
-                        f"times without success in {self.run_path}"
-                    )
-                if self.callback_timeout:
-                    self.callback_timeout(*self.callback_arguments)
+                self._log_kill_timeout_status()
+                self.run_timeout_callback()
```
Comment (oyvindeide) [inline src/ert/job_queue/job_queue_node.py:250]: Are we ever using a timeout callback?

Comment (eivindjahren) [reply src/ert/job_queue/job_queue_node.py:250]: Don't think so, but it made sense to not change the interface to the class in this refactoring.

File: src/ert/job_queue/job_queue_node.py [hunk 2]
```diff
@@ -134,7 +162,7 @@ def timed_out(self) -> bool:
     def submit_attempt(self) -> int:
         return self._get_submit_attempt()  # type: ignore
 
-    def refresh_status(self, driver: "Driver") -> JobStatusType:
+    def poll_driver_status(self, driver: "Driver") -> JobStatusType:
```
Comment (valentin-krasontovitsch) [inline src/ert/job_queue/job_queue_node.py:165]:
```suggestion
    def poll_queue_status(self, driver: "Driver") -> JobStatusType:
```

File: src/ert/job_queue/job_queue_node.py [hunk 3]
```diff
@@ -24,6 +24,34 @@
...
+    def __call__(
+        self,
+        elapsed_time: float,
+    ) -> int:
+        if (
+            self.time_sleep_seconds != self.max_sleep_seconds
+            and elapsed_time > self.time_until_longer_sleep_seconds
+        ):
+            self.time_sleep_seconds = self.max_sleep_seconds
+            self.use_random_sleep_offset = True
```
Comment (andreas-el) [inline src/ert/job_queue/job_queue_node.py:49]: This will never we false?

File: src/ert/job_queue/job_queue_node.py [hunk 4]
```diff
@@ -183,73 +215,86 @@ def runtime(self) -> float:
...
+            if self._start_time is not None:
+                elapsed_time = time.time() - self._start_time
+            else:
+                elapsed_time = 0.0
+            time.sleep(backoff(elapsed_time))
+            if self._exceeded_allowed_runtime():
                 self._kill(driver)
                 # We sometimes end up in a state where we are not able to kill it,
                 # so we end up flooding the logs with identical statements, so we
                 # check before we log.
```
Comment (valentin-krasontovitsch) [inline src/ert/job_queue/job_queue_node.py:249]: these comments need to be moved down to `_log_kill_timeout_status` if we want to keep them

**Review by andreas-el — APPROVED** [review]
LGTM

File: src/ert/job_queue/job_queue_node.py [hunk 5]
```diff
@@ -258,43 +303,52 @@ def _job_monitor(
...
                     )
-                    self._set_thread_status(ThreadStatus.READY)
-                    return
+                    self._transition_status(ThreadStatus.READY, current_status)
                 else:
-                    logger.error(
-                        f"Realization: {self.callback_arguments[0].iens} "
+                    self._transition_to_failure(
+                        message=f"Realization: {self.callback_arguments[0].iens} "
                         f"failed after reaching max submit with: {self._status_msg}"
```
Comment (valentin-krasontovitsch) [inline src/ert/job_queue/job_queue_node.py:320]: how do you feel about adding the number max submit in the log here?
```suggestion
                        f"failed after reaching max submit {max_submit} with: {self._status_msg}"
```

**Review by valentin-krasontovitsch — APPROVED** [review]
i haz comments. otherwise is good.

**Review by valentin-krasontovitsch — APPROVED** [review]
delish

**Pull request merged**